hadoop-ozone/client/src/test/java/org/apache/hadoop/ozone/client/TestReplicatedFileChecksumHelper.java

```diff
@@ -0,0 +1,223 @@
+import org.apache.hadoop.hdds.scm.XceiverClientFactory;
+import org.apache.hadoop.hdds.scm.XceiverClientGrpc;
+import org.apache.hadoop.hdds.scm.XceiverClientReply;
+import org.apache.hadoop.hdds.scm.pipeline.Pipeline;
+import org.apache.hadoop.hdds.scm.pipeline.PipelineID;
+import org.apache.hadoop.ozone.client.rpc.RpcClient;
+import org.apache.hadoop.ozone.om.helpers.OmKeyInfo;
+import org.apache.hadoop.ozone.om.helpers.OmKeyLocationInfo;
+import org.apache.hadoop.ozone.om.helpers.OmKeyLocationInfoGroup;
+import org.apache.hadoop.ozone.om.protocol.OzoneManagerProtocol;
+import org.apache.hadoop.util.DataChecksum;
+import org.apache.hadoop.util.Time;
+import org.apache.ratis.thirdparty.com.google.protobuf.ByteString;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import static org.apache.hadoop.hdds.protocol.datanode.proto.ContainerProtos.ChecksumType.CRC32;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit tests for ReplicatedFileChecksumHelper class.
+ */
+public class TestReplicatedFileChecksumHelper {
+
+  @Test
+  public void testEmptyBlock() throws IOException {
+    // test the file checksum of a file with an empty block.
+    RpcClient rpcClient = Mockito.mock(RpcClient.class);
+
+    OzoneManagerProtocol om = Mockito.mock(OzoneManagerProtocol.class);
+    when(rpcClient.getOzoneManagerClient()).thenReturn(om);
+
+    OmKeyInfo omKeyInfo = new OmKeyInfo.Builder()
+        .setVolumeName(null)
+        .setBucketName(null)
+        .setKeyName(null)
+        .setOmKeyLocationInfos(Collections.singletonList(
+            new OmKeyLocationInfoGroup(0, new ArrayList<>())))
+        .setCreationTime(Time.now())
+        .setModificationTime(Time.now())
+        .setDataSize(0)
+        .setReplicationConfig(new RatisReplicationConfig(
+            HddsProtos.ReplicationFactor.ONE))
+        .setFileEncryptionInfo(null)
+        .setAcls(null)
+        .build();
+
+    when(om.lookupKey(any())).thenReturn(omKeyInfo);
+
+    OzoneVolume volume = Mockito.mock(OzoneVolume.class);
+    when(volume.getName()).thenReturn("vol1");
+    OzoneBucket bucket = Mockito.mock(OzoneBucket.class);
+    when(bucket.getName()).thenReturn("bucket1");
+
+    ReplicatedFileChecksumHelper helper = new ReplicatedFileChecksumHelper(
+        volume, bucket, "dummy", 10, rpcClient);
+    helper.compute();
+    FileChecksum fileChecksum = helper.getFileChecksum();
+    assertTrue(fileChecksum instanceof MD5MD5CRC32GzipFileChecksum);
+    assertEquals(DataChecksum.Type.CRC32,
+        ((MD5MD5CRC32GzipFileChecksum)fileChecksum).getCrcType());
+  }
+
+  @Test
+  public void testOneBlock() throws IOException {
```

Comment:
Thanks for the good test case. Since this is primary patch I agree to keep it simple here and push this patch in asap. 

Adding a note here for including more test cases in follow-up jira tasks. I'd to to add more test cases with variable length values(-1, 0, >0), with more blocks.